components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java

```diff
@@ -124,6 +126,15 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) {
                 authenticationContext.addParameter(OAUTH2_ALLOWED_SCOPES, responseDTO.getScope());
                 authenticationContext.addParameter(OAUTH2_VALIDATE_SCOPE,
                         AuthConfigurationUtil.getInstance().isScopeValidationEnabled());
+                try {
+                    if (MDC.get(SERVICE_PROVIDER) != null) {
```

Comment:
From where the SERVICE_PROVIDER is set to MDC initially. If this is the only place, MDC.put(SERVICE_PROVIDER.. will never happen.

Reply:
Already fixed https://github.com/wso2-extensions/identity-carbon-auth-rest/blob/master/components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java#L152